components/script/dom/bindings/utils.rs

```diff
@@ -388,7 +389,7 @@ pub unsafe extern "C" fn resolve_global(
     let bytes = slice::from_raw_parts(ptr, length as usize);
 
     if let Some(init_fun) = InterfaceObjectMap::MAP.get(bytes) {
-        init_fun(cx, Handle::from_raw(obj));
+        init_fun(*cx, Handle::from_raw(obj));
```

Comment:
@jdm is this what you meant (in terms of function call changes)?

Reply:
In this case, you modified MAP to store functions that accept `script_runtime::JSContext`. `enumerate_global` and `resolve_global` either need to accept `cx: script_runtime::JSContext` (and pass it unmodified to `init_fun`) or `cx: *mut JSContext` (and pass it to `init_fun` by creating a new `script_runtime::JSContext` based on `cx`).